source/reference/data-types.txt

```diff
@@ -472,9 +516,117 @@ stored as type ``int``.
      }
    ]
 
+.. _timestamp-ex-default:
+
+Timestamp a New Document
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use ``Timestamp()`` without parameters to insert multiple documents
+using the default settings:
+
+.. code-block:: javascript
+
+   db.flights.insertMany(
+      [
+         { arrival: "true", ts: Timestamp() },
+         { arrival: "true", ts: Timestamp() },
+         { arrival: "true", ts: Timestamp() }
+      ]
+   )
+
+Run ``db.flights.find({})`` to see the timestamps. Notice that even
+though all three entries were stamped in the same second, the interval
+was incremented on each one. 
+
+.. code-block:: javascript
+   :copyable: false
+
+   [
+      {
+         _id: ObjectId("6114216907d84f5370391919"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1628709225, i: 1 })
+      },
+      {
+         _id: ObjectId("6114216907d84f537039191a"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1628709225, i: 2 })
+      },
+      {
+         _id: ObjectId("6114216907d84f537039191b"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1628709225, i: 3 })
+      }
+   ]
+
+.. _timestamp-ex-custom:
+
+Create a Custom Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use custom parameters to insert multiple documents with a specific
+``Timestamp``. 
 
-Type Checking, ``$type()``
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+This operation inserts three documents into the ``flights`` collection
+and uses the :term:`UNIX epoch <unix epoch>` value ``1627811580`` to
+set the ``ts`` times to 9:53 GMT on August 1, 2021.
+
+.. code-block:: javascript
+
+   db.flights.insertMany(
+      [
+         { arrival: "true", ts: Timestamp(1627811580, 10) },
+         { arrival: "true", ts: Timestamp(1627811580, 20) },
+         { arrival: "true", ts: Timestamp(1627811580, 30) }
+      ]
+   )
+
+The resulting documents look like this:
+
+.. code-block:: javascript
+   :copyable: false
+
+   [
+      {
+         _id: ObjectId("6123d8315e6bba6f61a1031c"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1627811580, i: 10 })
+      },
+      {
+         _id: ObjectId("6123d8315e6bba6f61a1031d"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1627811580, i: 20 })
+      },
+      {
+         _id: ObjectId("6123d8315e6bba6f61a1031e"),
+         arrival: 'true',
+         ts: Timestamp({ t: 1627811580, i: 30 })
+      }
+   ]
+
+.. _timestamp-ex-extract:
+
+Extract the Date from a System Generated Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB uses the Timestamp() constructor internally when it creates an
+:ref:`ObjectId`. You can use the :method:`ObjectId.getTimestamp()`
```

Comment:
The point of my earlier comment was that Timestamp and ObjectId are entirely unrelated types, other than that both use UNIX timestamps (i.e. the first sentence here is inaccurate), and that I think it would be confusing for readers if the docs imply that they are related in some way.

Reply:
I've removed the example, would you have a look? Thanks!